contracts/Library/LibraryLock.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-3.0
```

Comment:
I'm not sure I completely understand the purpose of this contract.

Reply:
To make sure that no one can call functions from the implementation (logic), also to prevent damage to the logic Contract, it's recommended to restrict permissions for any potentially damaging functions

Reply:
looking at the code, and at how you use it, i don't see how it prevents direct calls on the implementation... As long as `initialized` is set as `true` you can make direct calls

Reply:
Yes it is set to **true** for the proxy not for the logic

Reply:
i don't think it is the case here, i don't see what prevents you to call the function that sets `initialized` to `true` on the logic while you can call it from the proxy?

Reply:
Normally we shouldn't have to interact with the implementation, we only do it to initialize and prevent others from using the contract (and possibly change its nature).
To answer your question, If you call it from the proxy, it will be set to `true` only for this proxy, don't hesitate to experiment on your own